0006-ffmul-revised.md

```diff
@@ -1302,7 +1296,7 @@ Compute and constrain the native modulus values, which are used to check the con
 - $r_n = 2^{2\ell} \cdot r_2 + r_{01} \mod n$
 - $f_n = 2^{2\ell} \cdot f_2 + 2^{\ell} \cdot f_1 + f_0 \mod n$
 
-Actually, there is no need to store in the coefficients the limbs of $f$, but only $(f'_0, f'_1, f'_2)$ and $f_2$. Then the native constrain uses the negated limbs of `f` instead.
+Actually, there is no need to store in the coefficients the limbs of $f$, but only $(f'_0, f'_1, f'_2)$ and $f_2$. Then the native constrain uses the negated limbs of `f` instead and subtracts the additional $2^{3\ell}q_n$ term.
```

Comment:
This crucial detail of how to implement the native constraint is nearly at the end of the appendix. This is not good.

Reply:
Actually we also spell out this detail in the gate layout section (main part of the RFC):

> The limbs of $f' = 2^{3\ell} - f$, which feature in a few of the constraints, are embedded into the gate as coefficients. The $\text{(C1)}$ constraint features $f$, but the gate implementation expands it out as $f = 2^{3\ell} - f' = 2^{3\ell} - (f_0' + 2^\ell f_1' + 2^{2\ell} f_2')$ to be able to use the same coefficients